sdk/src/test/scala/com/horizen/account/state/CertificateKeyRotationMsgProcessorTest.scala

```diff
@@ -0,0 +1,191 @@
+
+  val masterKeyType: KeyRotationProofType.Value = KeyRotationProofType.MasterKeyRotationProofType
+  val signingKeyType: KeyRotationProofType.Value = KeyRotationProofType.SigningKeyRotationProofType
+
+  @Before
+  def setUp(): Unit = {
+  }
+
+  def generateKeys(n: Int): immutable.Seq[SchnorrSecret] = {
+    (0 to n).map(_ => keyGenerator.generateSecret(Random.nextDouble().toByte.toByteArray))
+  }
+
+  def getDefaultMessage(opCode: Array[Byte], arguments: Array[Byte], nonce: BigInteger, value: BigInteger = bigIntegerZero): Message = {
+    val data = Bytes.concat(opCode, arguments)
+    new Message(
+      new AddressProposition(origin),
+      new AddressProposition(contractAddress), // to
+      bigIntegerZero, // gasPrice
+      bigIntegerZero, // gasFeeCap
+      bigIntegerZero, // gasTipCap
+      bigIntegerZero, // gasLimit
+      value,
+      nonce,
+      data,
+      false)
+  }
+
+  def randomNonce: BigInteger = randomU256
+
+
+  def createSenderAccount(view: AccountStateView, amount: BigInteger = BigInteger.ZERO): Unit = {
+    if (!view.accountExists(origin)) {
+      view.addAccount(origin, randomHash)
+
+      if (amount.signum() == 1) {
+        view.addBalance(origin, amount)
+      }
+    }
+  }
+
+  @Test
+  def testMethodIds(): Unit = {
+    //The expected methodIds were calcolated using this site: https://emn178.github.io/online-tools/keccak_256.html
+    assertEquals("Wrong MethodId for SubmitKeyRotationReqCmdSig", "aa5a0d75", CertificateKeyRotationMsgProcessor.SubmitKeyRotationReqCmdSig)
+  }
+
+
+  @Test
+  def testSubmitKeyRotation(): Unit = {
+    val oldMasterKey = generateKeys(1)
+    val oldSigningKey = generateKeys(1)
+    val newMasterKey = generateKeys(1).head
+
+    val keyRotationProof = KeyRotationProof(
+      masterKeyType,
+      0,
+      newMasterKey.publicImage(),
+      oldSigningKey.head.sign(newMasterKey.getPublicBytes.take(32)),
+      oldMasterKey.head.sign(newMasterKey.getPublicBytes.take(32))
+    )
+
+    usingView(certificateKeyRotationMsgProcessor) { view =>
+
+      certificateKeyRotationMsgProcessor.init(view)
+      when(mockNetworkParams.signersPublicKeys).thenReturn(oldSigningKey.map(_.publicImage()))
+      when(mockNetworkParams.masterPublicKeys).thenReturn(oldMasterKey.map(_.publicImage()))
+
+      val cmdInput = SubmitKeyRotationCmdInput(keyRotationProof, newMasterKey.sign(newMasterKey.getPublicBytes.take(32)))
+      val data: Array[Byte] = cmdInput.encode()
+      val msg = getMessage(to = contractAddress, data = BytesUtils.fromHexString(SubmitKeyRotationReqCmdSig) ++ data, nonce = randomNonce)
+
+      withGas {
+        certificateKeyRotationMsgProcessor.process(msg, view, _, defaultBlockContext)
+      }
+
+      val masterKeyHistory = certificateKeyRotationMsgProcessor.getKeysRotationHistory(masterKeyType, 0, view)
+      assertEquals("Key Rotation history is not updated", List(0), masterKeyHistory.epochNumbers)
```

Comment:
Consider to use List(defaultBlockContext.withdrawalEpochNumber) instead of List(0), for clarity and because the test won't fail in case defaultBlockContext.withdrawalEpochNumber changes